### ISSUE TITLE
reduce the tree

### ISSUE THREAD

**Issue by sushovan-mryum** [tckerr/walk#3]
Great library exactly what I was looking for...

Is there any way we can reduce the tree.
Something like here: https://jrsinclair.com/articles/2019/functional-js-traversing-trees-with-recursive-reduce/

Basically would like to return a new transformed/filtered tree

**Comment by tckerr** [comment]
Good suggestion. I believe this type of behavior is possible to achieve through usage of the standard walkjs toolkit, although it's also a good candidate for a built-in utility.

In lieu of that, you could do a reduce manually by keeping the accumulator outside of the walk:

```javascript
import {apply} from 'walkjs'

let myTree = {}
let acc = 0;

apply(myTree, node => acc++):
```

Filtering a tree would be slightly harder in that you need to construct a new tree as your accumulator. However, this is exactly how the built-in function `deepCopy` works. It would be easy to write your own version of this which adds filters. See the implementation here: https://github.com/tckerr/walk/blob/master/src/utils/deepCopy.ts.

**Comment by tckerr** [comment]
Closing this issue, as I've implemented a reduce function in version `3.2.0+`: https://github.com/tckerr/walk#reduce
